basic_crud/basic_crud/basic_crud.py

```diff
@@ -0,0 +1,202 @@
+"""Welcome to Reflex! This file outlines the steps to create a basic app."""
+import asyncio
+import json
+from basic_crud.api import product_router
+from basic_crud.model import Product
+from rxconfig import config
+import httpx
+
+import reflex as rx
+
+docs_url = "https://reflex.dev/docs/getting-started/introduction"
+filename = f"{config.app_name}/{config.app_name}.py"
+
+
+DEFAULT_BODY = """{
+    "code":"",
+    "label":"",
+    "image":"/favicon.ico",
+    "quantity":0,
+    "category":"",
+    "seller":"",
+    "sender":""
+}"""
+
+URL_OPTIONS = {
+    "GET": "products",
+    "POST": "products",
+    "PUT": "products/{pr_id}",
+    "DELETE": "products/{pr_id}",
+}
+
+
+class State(rx.State):
+    """The app state."""
+
+    products: list[Product]
+    _db_updated: bool = False
+
+    def load_product(self):
+        with rx.session() as session:
+            self.products = session.query(Product).all()
+        yield State.reload_product
+
+    @rx.background
+    async def reload_product(self):
```

Comment:
this one seems kind of redundant to `load_product`... is it intended to continuously reload? Currently it seems like it only reloads if the user makes some change, but what if multiple users are working with the data, it wouldn't update to show the other user's changes.

I think if `reload_product` is going to be here, the code should ensure that only one copy of the task is running and it should always reload on some interval.

Down below where `_db_updated` is set `True` when a request is successful, i think the code should just `yield State.load_product` to reload the new data in response to user action. This assumes that there is a guard in place to prevent `reload_product` from running concurrently with itself.

Reply:
This one is a workaround, I wanted a "clean" way to trigger event when the API is called (so not when we receive the answer, as the request could come from another REST client) but 0.2.9 doesn't offer this yet.

(I could copy the broadcast example of chatroom, but it's not very understandable for user, so I'd rather hold off and clean it later)